### DISCUSSION TITLE
RelConnectsRigidCross aplicable also on Ribs (StructuralCurveMemberRib)

### DISCUSSION THREAD

**Discussion by sonkaste17** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#240] [Ideas]
Currently, RelConnectsRigidCross (Cross-link) can be applied only on two StructuralCurveMembers and not to StructuralCurveMemberRib. Input on ribs is logical usage, so it would be great to have the option to apply the RelConnectsRigidCross on the ribs too or even the combination of a rib and an ordinary member.

![image](https://user-images.githubusercontent.com/81751485/199519904-81329957-c2ca-445a-abc4-49833a1b4e2c.png)

**Comment by sonkaste17** [comment]
@msalak-scia Can you elaborate on a proposal, please?

**Reply by MartinSalak** [reply]
I am on it

**Comment by MartinSalak** [comment]
Hello SAF community, here is a proposal for changes in **RelConnectsRigidCross**. Previously RelConnectsRigidCross was only applicable to beams. With this new change it can be also applied to ribs, so you can connect beam with a rib.

As we don’t have unique names across the whole SAF model (only for one object) it is not possible to assign 1D members and Ribs in one row. The proposal below adds one new row where ribs can be referenced.

It is always necessary to define exactly two members connected by rigid cross. This means two 1D members, one 1D member and one rib, or two ribs.

Changed rows:
| Property | Example | Required | Definition |
| --- | --- | --- | --- |
| 1D Members | B1; B2 | yes,<br>if connection between 1D member(s) and other entity is desired,<br>otherwise no | Reference to [StructuralCurveMember]( ../structural-analysis-elements/structuralcurvememberrib.md)<br>Exactly two members from among 1D members and Ribs must be defined; either two ribs, two 1D members, or one of each. |

New rows:
| Property | Example | Required | Definition |
| --- | --- | --- | --- |
| Ribs | RB1; RB2 | yes,<br>if connection between rib(s) and other entity is desired,<br>otherwise no | Reference to [StructuralCurveMemberRib]( ../structural-analysis-elements/structuralcurvememberrib.md)<br>Exactly two members from among 1D members and Ribs must be defined; either two ribs, two 1D members, or one of each. |

**Reply by MartinSalak** [reply]
@dschuermans Could you please look at it? Does it seem ok to you? I am a little puzzled about the required values. There must be always two input values, but it can be two 1D members and no Ribs, one 1D member and one rib or two ribs and no 1D Member. Is it OK to describe it in the description and the values not being required? Or do we need new row which would say which one of these combination it is? 
(it is not possible to skip these rows, at least one of them must contain some information)

**Reply by dschuermans** [reply]
> @dschuermans Could you please look at it? Does it seem ok to you? I am a little puzzled about the required values. There must be always two input values, but it can be two 1D members and no Ribs, one 1D member and one rib or two ribs and no 1D Member. Is it OK to describe it in the description and the values not being required? Or do we need new row which would say which one of these combination it is? (it is not possible to skip these rows, at least one of them must contain some information)

There's some other object as well that has these kind of references (to node, beam and surfaces I think) where you can make some combo of all the different types: 
https://www.saf.guide/en/stable/supports-and-hinges/relconnectsrigidmember.html

Isn't that the same behavior?

**Reply by Heroog** [reply]
Is there a reason why we don't have the names unique across the whole model? Not sure why we did't do that. Seems to me logic and I suppose most users/vendors are doing it by nature. If we say name must be unique then you just have to check if the name is from CurveMember or CurveMemberRib.

**Reply by dschuermans** [reply]
> Is there a reason why we don't have the names unique across the whole model? Not sure why we did't do that. Seems to me logic and I suppose most users/vendors are doing it by nature. If we say name must be unique then you just have to check if the name is from CurveMember or CurveMemberRib.

@Heroog there's been a discussion about this in detail:
https://github.com/StructuralAnalysisFormat/StructuralAnalysisFormat-Doc/discussions/7#discussioncomment-1050885

**Reply by Heroog** [reply]
and what was final conclusion? We didn't do anything with it?

**Reply by dschuermans** [reply]
> and what was final conclusion? We didn't do anything with it?

https://github.com/StructuralAnalysisFormat/StructuralAnalysisFormat-Doc/discussions/7#discussioncomment-1064979

It would be a breaking changing, warranting a major version bump and we simply do not have the resources at this point to make those changes to the SDK

**Reply by MartinSalak** [reply]
> Isn't that the same behavior?

You are right, thanks. I didn't notice this one.
